components/server/ee/src/user/user-deletion-service.ts

```diff
@@ -10,38 +10,59 @@ import { SubscriptionService } from "@gitpod/gitpod-payment-endpoint/lib/account
+            const now = new Date();
+            const subscriptions = await this.subscriptionService.getNotYetCancelledSubscriptions(user, now.toISOString());
             for (const subscription of subscriptions) {
-                const planId = subscription.planId!;
-                if (Plans.isFreeNonTransientPlan(planId)) {
-                    // only delete those plans that are persisted in the DB
-                    await this.subscriptionService.unsubscribe(user.id, now, planId);
-                } else if (Plans.isFreePlan(planId)) {
-                    // we do not care about transient plans
-                    continue;
-                } else {
-                    // cancel Chargebee subscriptions
-                    const subscriptionId = subscription.uid;
-                    const chargebeeSubscriptionId = subscription.paymentReference!;
-                    await this.chargebeeService.cancelSubscription(chargebeeSubscriptionId, { userId: user.id }, { subscriptionId, chargebeeSubscriptionId });
+                try {
+                    const planId = subscription.planId!;
+                    const paymentReference = subscription.paymentReference;
+                    if (Plans.isFreeNonTransientPlan(planId)) {
+                        // only delete those plans that are persisted in the DB
+                        await this.subscriptionService.unsubscribe(user.id, now.toISOString(), planId);
+                    } else if (Plans.isFreePlan(planId)) {
+                        // we do not care about transient plans
+                        continue;
+                    } else {
+                        if (!paymentReference) {
+                            const teamSlots = await this.teamSubscriptionService.findTeamSubscriptionSlotsByAssignee(id);
+                            teamSlots.forEach(async ts => await this.teamSubscriptionService.deactivateSlot(ts.teamSubscriptionId, ts.id, now))
+                        } else if (paymentReference.startsWith("github:")) {
+                            throw new Error("Cannot delete user subscription from GitHub")
+                        } else {
+                            // cancel Chargebee subscriptions
+                            const subscriptionId = subscription.uid;
+                            const chargebeeSubscriptionId = subscription.paymentReference!;
+                            await this.chargebeeService.cancelSubscription(chargebeeSubscriptionId, { userId: user.id }, { subscriptionId, chargebeeSubscriptionId });
+                        }
+                    }
+                } catch (error) {
+                    errors.push(error);
+                    log.error("Error cancelling subscription", error, { subscription })
                 }
             }
         }
 
-        return super.deleteUser(id);
+        await super.deleteUser(id);
+        if (errors) {
+            throw new Error(errors.join("\n"))
```

Comment:
why throwing the errors here? We still deleted the user and already logged it.

Reply:
The idea was to give feedback to the caller, e.g. the admin dashboard.

In particular in the case of GitHub subscription, it may be worthwhile to notify the customer that their subscription will still be charged as we have no means to cancel it.